.github/workflows/everything.yml

```diff
@@ -131,6 +131,10 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           path: source
+      - name: ccache
```

Comment:
@scottwittenburg We already provide ccache in the docker images, why downloading it again with this ccache action. I can see that this also setup some ccache settings. 
Concretely:
```
    await execBash(`ccache --set-config=cache_dir='${path.join(ghWorkSpace, '.ccache')}'`);
    await execBash(`ccache --set-config=max_size='${maxSize}'`);
    await execBash(`ccache --set-config=compression=true`);
```

It might be better doing this ourselves and avoid this dependency. Note that in ccache all these settings can be also setup with a env variable CCACHE_...`

Reply:
> why downloading it again with this ccache action?

This action only installs `ccache` if it doesn't exist already, see [here](https://github.com/hendrikmuhs/ccache-action/blob/53e53642769c4b03a7881a7283314d1f364134ec/src/restore.ts#L169).  So it's handling it for us on the mac builds (where we'll need to manage it ourselves in some script otherwise), but not on the linux ones.

> It might be better doing this ourselves and avoid this dependency

We already depend on a few other actions, I'm not sure I understand the reasoning to avoid this one, since it makes things pretty easy.  See the [actions.yml](https://github.com/hendrikmuhs/ccache-action/blob/53e53642769c4b03a7881a7283314d1f364134ec/action.yml#L11) for the configuration options it exposes, we could set the max size if you think it should be different.  It seems we cannot change the cache dir or compression, but I don't know if we would want to change those anyway.

Having said that, I'll leave the decision up to you.

Reply:
>  It seems we cannot change the cache dir or compression, but I don't know if we would want to change those anyway.

Yes not need to change it a the moment.

The reasoning is for getting rid of this, is that using ccache without this action is not really harder than using this github-action. Also configuring it directly can let us in the future to further change options if needed. You can try to setup the options that it sets through env vars and it should yield the same result. Actually compression is true by default so you can skip it.

Reply:
Ok, let me try it without the action.